spec/services/courses/age_range_calculator_service_spec.rb

```diff
@@ -0,0 +1,107 @@
+describe Courses::AgeRangeCalculatorService do
+  let(:course) { create :course, provider: provider, changed_at: Time.now + 60 }
+  let(:provider) { build(:provider) }
+  subject { Courses::AgeRangeCalculatorService.determine_age_range_from(course) }
+
+  context "age_range_in_years" do
+    context "3 to 11" do
+      context "3_to_7" do
+        it { is_expected.to eq("P") }
+      end
+      context "5_to_11" do
+        let(:course) {
+          create :course,
+                 age_range_in_years: "5_to_11",
+                 provider: provider
+        }
+        it { is_expected.to eq("P") }
+      end
+      context "7_to_11" do
+        let(:course) {
+          create :course,
+                 age_range_in_years: "7_to_11",
+                 provider: provider
+        }
+        it { is_expected.to eq("P") }
+      end
+    end
+    context "7 - 14" do
+      let(:course) {
+        create :course,
+               age_range_in_years: "7_to_14",
+               provider: provider
+      }
+      it { is_expected.to eq("M") }
+    end
+    context "11 - 19" do
+      context "11_to_16" do
+        let(:course) {
+          create :course,
+                 age_range_in_years: "11_to_16",
+                 provider: provider
+        }
+        it { is_expected.to eq("S") }
+      end
+      context "11_to_18" do
+        let(:course) {
+          create :course,
+                 age_range_in_years: "11_to_18",
+                 provider: provider
+        }
+        it { is_expected.to eq("S") }
+      end
+      context "14_to_19" do
+        let(:course) {
+          create :course,
+                 age_range_in_years: "14_to_19",
+                 provider: provider
+        }
+        it { is_expected.to eq("S") }
+      end
+    end
+
+    context "is an unknown value" do
+      let(:course) {
+        create :course,
+               age_range_in_years: "21_to_25",
+               provider: provider,
+               is_send: true
+      }
+      it { is_expected.to eq("O") }
+    end
+
+    context "is nil and level is primary" do
+      let(:course) {
+        create :course,
+               age_range_in_years: nil,
+               level: "primary",
+               provider: provider,
+               is_send: true
+      }
+
+      it { is_expected.to eq("P") }
+    end
+    context "is nil and level is secondary" do
+      let(:course) {
+        create :course,
```

Comment:
maybe add 

create :course,
               age_range_in_years: 21_to_25,
               level: "further_education"

as well

Reply:
test case added for `age_range_in_years == nil` and `level == [some unknown value]`

Reply:
Success criteria has since been updated (and simplified 😄), the extra test cases are no longer required